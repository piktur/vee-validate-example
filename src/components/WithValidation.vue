<template>
  <ValidationProvider
    rules="required|email"
    name="email"
    mode="lazy"
    slim
    vid="email">
    <template v-slot="{ ariaInput, errors }">
      <VTextField
        v-bind="ariaInput"
        v-model.lazy="form.email"
        :error-messages="errors"
        label="Email"
        hint="required|email"
        name="email" />
    </template>
  </ValidationProvider>
</template>

<script lang="ts">
import Vue from 'vue';
import { VTextField } from 'vuetify/lib'
import { extend, ValidationProvider } from 'vee-validate'
import { required, email } from 'vee-validate/dist/rules'

extend('required', required)
extend('email', email)

export default Vue.extend({
  name: 'WithValidation',
  components: { VTextField, ValidationProvider },
  data: () => ({
    form: {
      email: ''
    }
  }),
});
</script>
