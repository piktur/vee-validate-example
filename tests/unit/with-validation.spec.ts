import { mount } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import WithValidation from '@/components/WithValidation.vue'

describe('WithValidation.vue', () => {
  const wrapper = mount(WithValidation)

  const textField = wrapper.find('.v-text-field')
  const input = textField.find('input')

  const vnode = textField.vm.$vnode
  const componentOptions = vnode.componentOptions
  // @ts-ignore
  const listeners = vnode.componentOptions.listeners


  it('applies listeners correctly', () => {
    expect(listeners).toMatchObject({
      input: expect.any(Function),
      change: expect.any(Function),
      blur: expect.any(Function),
    })
  })

  it('displays validation message', async () => {
    input.setValue('invalid')
    input.trigger('change') // either event should trigger validation
    input.trigger('blur')

    await flushPromises()
    await wrapper.vm.$nextTick()

    expect(textField.find('.v-messages__message').text()).toMatchSnapshot()
  })
})
