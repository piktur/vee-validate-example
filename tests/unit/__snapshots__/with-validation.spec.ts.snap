// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithSimpleValidation.vue displays validation message 1`] = `"email is not valid."`;
